packages/blade/src/components/Chip/Chip.tsx

```diff
@@ -0,0 +1,237 @@
+
+const _Chip: React.ForwardRefRenderFunction<BladeElementRef, ChipProps> = (
+  { isDisabled, value, children, icon: Icon, testID, ...styledProps },
+  ref,
+) => {
+  const { theme } = useTheme();
+  const { matchedDeviceType } = useBreakpoint({ breakpoints: theme.breakpoints });
+  const groupProps = useChipGroupContext();
+  const isInsideGroup = !isEmpty(groupProps);
+  const [isPressed, setIsPressed] = React.useState(false);
+
+  if (__DEV__) {
+    if (!isInsideGroup) {
+      throwBladeError({
+        moduleName: 'Chip',
+        message:
+          '<Chip /> component should only be used within the context of a <ChipGroup /> component',
+      });
+    }
+  }
+
+  const _isDisabled = isDisabled ?? groupProps?.isDisabled;
+  const _name = groupProps?.name;
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+  const _isChecked = groupProps?.state?.isChecked(value!);
+  // Check if the defaultValue property of groupProps is undefined
+  const defaultChecked =
+    typeof groupProps?.defaultValue === 'undefined'
+      ? undefined // If undefined, defaultChecked is also undefined
+      : groupProps?.defaultValue?.includes(value as string); // If multiple selection, check if value is in defaultValue array
+  const useChip = groupProps?.selectionType === 'single' ? useRadio : useCheckbox;
+  const _size = groupProps?.size || 'small';
+  const _intent = groupProps?.intent;
+
+  const handleChange: OnChange = ({ isChecked, value }) => {
+    if (isChecked) {
+      groupProps?.state?.addValue(value!);
+    } else {
+      groupProps?.state?.removeValue(value!);
+    }
+  };
+
+  const { state, inputProps } = useChip({
+    defaultChecked,
+    isChecked: _isChecked,
+    isDisabled: _isDisabled,
+    name: _name,
+    value,
+    onChange: handleChange,
+  });
+
+  const handlePointerPressedIn = React.useCallback(() => {
+    if (_isDisabled) return;
+    setIsPressed(true);
+  }, [_isDisabled]);
+
+  const handlePointerPressedOut = React.useCallback(() => {
+    if (_isDisabled) return;
+    setIsPressed(false);
+  }, [_isDisabled]);
+
+  const handleKeyboardPressedIn = React.useCallback(
+    (e: React.KeyboardEvent) => {
+      if (_isDisabled) return;
+      if (e.key === ' ') {
+        setIsPressed(true);
+      }
+    },
+    [_isDisabled],
+  );
+
+  const handleKeyboardPressedOut = React.useCallback(
+    (e: React.KeyboardEvent) => {
+      if (_isDisabled) return;
+      if (e.key === ' ') {
+        setIsPressed(false);
+      }
+    },
+    [_isDisabled],
+  );
+
+  const chipTextColor =
+    chipColorTokens.text[_isDisabled ? 'disabled' : _isChecked && _intent ? _intent : 'unchecked'];
+  const chipBackgroundColor =
+    chipColorTokens.background[_isChecked && _intent ? _intent : 'unchecked'][
+      _isDisabled ? 'disabled' : 'default'
+    ];
+  const chipBorderColor =
+    chipColorTokens.border[_isChecked && _intent ? _intent : 'unchecked'][
+      _isDisabled ? 'disabled' : 'default'
+    ];
```

Comment:
phew, quite a lot of ternaries. 1 level deep ternaries are fine but this is a bit hard to read. 

Can we simplify? 

```jsx
  let textVariant = 'unchecked';
  if (_isChecked && _intent) {
    textVariant = _intent;
  }
  if (_isDisabled) {
    textVariant = 'disabled';
  }
  const chipTextColor = chipColorTokens.text[textVariant];

  let intentVariant = 'unchecked';
  const stateVariant = _isDisabled ? 'disabled' : 'default';
  if (_isChecked && _intent) {
    intentVariant = _intent;
  }
  const chipBackgroundColor = chipColorTokens.background[intentVariant][stateVariant];
  const chipBorderColor = chipColorTokens.border[intentVariant][stateVariant];
```

Reply:
Yes, updated 👍🏻